ukvm/ukvm_hv_kvm_aarch64.c

```diff
@@ -370,16 +370,19 @@ void ukvm_hv_vcpu_loop(struct ukvm_hv *hv)
                 errx(1, "Invalid guest mmio access: mmio=0x%llx", run->mmio.phys_addr);
 
             int nr = UKVM_HYPERCALL_NR(run->mmio.phys_addr);
+            ukvm_gpa_t gpa = mmio_read32(run->mmio.data);
 
-            /* Guest has halted the CPU, this is considered as a normal exit. */
-            if (nr == UKVM_HYPERCALL_HALT)
-                return;
+            /* Guest has halted the CPU. */
+            if (nr == UKVM_HYPERCALL_HALT) {
```

Comment:
Please do the declaration and load of `gpa` at the start of this `if` block, and keep the original declaration and load of `gpa` below, before the call to `fn(...)`.

Reply:
changed